### PR TITLE
Making singlefile apps work when wrapped in an OSX universal binary

### PR DESCRIPTION
**Pull request by VSadov**
The main idea is:
- reader checks for FAT/FAT64 magic numbers to detect the case when a singlefile app is embedded in a universal binary. 
- parse FAT/FAT64 headers to figure the offset within the outer file.
- remember the offset. 
- read singlefile metadata as-is, without adjustments. 
- adjust offsets by the `offset_in_file` in a few places where we read or probe for the data embedded in the singlefile app.

Alternatively we could adjust all offsets as we read them, but that is more complicated since we rely on direct reading of C++ structs. Also in a few places it would be ambiguous whether we work with already adjusted offsets or not.
It seems more consistent to internally keep all offsets as in the file and adjust, if needed, when passing to uses like `read` and `map`.

### CODE REVIEW

File: src/native/corehost/bundle/reader.h [hunk 1]
```diff
@@ -8,6 +8,39 @@
...
+    fat_arch_type* arch_list = (fat_arch_type*)(addr + sizeof(uint32_t) * 2);
+    for (int i = 0; i < nfat_arch; i++)
+    {
+        if (OSSwapBigToHostInt32((uint32_t)arch_list[i].cputype) == TARGET_CPU_TYPE)
+        {
+            return &arch_list[i].offset;
+        }
+    }
+
+    trace::error(_X("Couldn't find offset un an universal fat binary."));
```
Comment (vitek-karas) [inline src/native/corehost/bundle/reader.h:38]:
```suggestion
    trace::error(_X("Couldn't find offset in an universal fat binary."));
```

**Review by vitek-karas — COMMENTED** [review]
Looks good

**Comment by VSadov** [comment]
Added test scenarios. (also verified locally that the tests do fail if the new functionality is not enabled)

**Comment by VSadov** [comment]
I think this PR is ready for review/merge.

File: src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.Bundle.Tests/BundleAndRun.cs [hunk 2]
```diff
@@ -46,6 +46,21 @@ private void CheckFileNotarizable(string path)
...
+        {
+            string fatApp = path + ".fat";
+            string arch = BundleHelper.GetTargetArch(rid) == Architecture.Arm64 ? "arm64" : "x86_64";
+
+            Command.Create("lipo", $"-create -arch {arch} {path} -output {fatApp}")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should()
+                .Pass();
```
Comment (vitek-karas) [inline src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.Bundle.Tests/BundleAndRun.cs:59]: I think it's worth adding a comment that this creates a FAT binary with only one MACH-O image in it. So not truly a universal binary. But for the purposes of testing this should be good enough.

**Review by vitek-karas — APPROVED** [review]

**Comment by VSadov** [comment]
Thanks!

**Pull request merged**